pkg/services/blockchain/eth/resolver.go

```diff
@@ -30,16 +39,59 @@ type Resolver struct {
 
 	contractAddress string
 	chainID         int
+	walletKey       string
 }
 
+type AuthData struct {
+	TypedData apitypes.TypedData
+	Signature string
+	Address   string
+}
+
+const (
+	secp256k1VValue = 27
+)
+
 var (
 	gistNotFoundException     = "execution reverted: Root does not exist"
 	identityNotFoundException = "execution reverted: Identity does not exist"
 	stateNotFoundException    = "execution reverted: State does not exist"
 )
 
+var IdentityStateAPITypes = apitypes.Types{
+	"IdentityState": []apitypes.Type{
+		{Name: "timestamp", Type: "uint256"},
+		{Name: "id", Type: "uint256"},
+		{Name: "state", Type: "uint256"},
+		{Name: "replacedAtTimestamp", Type: "uint256"},
+	},
+	"EIP712Domain": []apitypes.Type{
+		{Name: "name", Type: "string"},
+		{Name: "version", Type: "string"},
+		{Name: "chainId", Type: "uint256"},
+		{Name: "verifyingContract", Type: "address"},
+	},
+}
+
+var GlobalStateAPITypes = apitypes.Types{
+	"GlobalState": []apitypes.Type{
+		{Name: "timestamp", Type: "uint256"},
+		{Name: "idType", Type: "bytes2"},
+		{Name: "root", Type: "uint256"},
+		{Name: "replacedAtTimestamp", Type: "uint256"},
+	},
+	"EIP712Domain": []apitypes.Type{
+		{Name: "name", Type: "string"},
+		{Name: "version", Type: "string"},
+		{Name: "chainId", Type: "uint256"},
+		{Name: "verifyingContract", Type: "address"},
+	},
+}
+
+var TimeStamp = TimeStampFn
+
 // NewResolver create new ethereum resolver.
-func NewResolver(url, address string) (*Resolver, error) {
+func NewResolver(url, address, walletKey string) (*Resolver, error) {
```

Comment:
As I understand walletKey is optional, and by setting it in NewResolver function as parameter we brake public interface. 

Let's follow options approach [example](https://github.com/iden3/go-schema-processor/blob/b2b672961cfd2a9613ec1a60a0b82395ac6923b0/loaders/document_loader.go#L67)

Reply:
Done